// _vant@0.11.5@vant/lib/search/index.js
'use strict';

exports.__esModule = true;

var _components;

var _icon = require('../icon');

var _icon2 = _interopRequireDefault(_icon);

var _clickoutside = require('../utils/clickoutside');

var _clickoutside2 = _interopRequireDefault(_clickoutside);

var _locale = require('../locale');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  render: function render() {
    var _vm = this;var _h = _vm.$createElement;var _c = _vm._self._c || _h;return _c('div', { staticClass: "van-search", class: { 'van-search--show-action': _vm.showAction }, style: { 'background-color': _vm.background } }, [_c('div', { directives: [{ name: "clickoutside", rawName: "v-clickoutside", value: _vm.onClickoutside, expression: "onClickoutside" }], staticClass: "van-search__input-wrap" }, [_c('van-icon', { attrs: { "name": "search" } }), _c('input', { directives: [{ name: "refocus", rawName: "v-refocus", value: _vm.focusStatus, expression: "focusStatus" }], staticClass: "van-search__input", attrs: { "type": "search", "placeholder": _vm.placeholder }, domProps: { "value": _vm.value }, on: { "input": _vm.onInput, "focus": _vm.onFocus, "keypress": function keypress($event) {
          if (!('button' in $event) && _vm._k($event.keyCode, "enter", 13, $event.key)) {
            return null;
          }$event.preventDefault();_vm.onSearch($event);
        } } }), _c('van-icon', { directives: [{ name: "show", rawName: "v-show", value: _vm.isFocus, expression: "isFocus" }], attrs: { "name": "clear" }, on: { "click": _vm.onClean } })], 1), _vm.showAction ? _c('div', { staticClass: "van-search__action" }, [_vm._t("action", [_c('div', { staticClass: "van-search__action-text", on: { "click": _vm.onBack } }, [_vm._v(_vm._s(_vm.$t('cancel')))])])], 2) : _vm._e()]);
  },

  name: 'van-search',

  mixins: [_locale.i18n],

  components: (_components = {}, _components[_icon2.default.name] = _icon2.default, _components),

  props: {
    value: String,
    showAction: Boolean,
    placeholder: String,
    background: {
      type: String,
      default: '#f2f2f2'
    }
  },

  data: function data() {
    return {
      isFocus: false,
      focusStatus: false
    };
  },


  directives: {
    Clickoutside: _clickoutside2.default,
    refocus: {
      update: function update(el, state) {
        if (state.value) {
          el.focus();
        }
      }
    }
  },

  methods: {
    onFocus: function onFocus() {
      this.isFocus = true;
    },
    onInput: function onInput(event) {
      this.$emit('input', event.target.value);
    },


    // refocus after click close icon
    onClean: function onClean() {
      var _this = this;

      this.$emit('input', '');
      this.focusStatus = true;

      // ensure refocus can work after click clean icon
      this.$nextTick(function () {
        _this.focusStatus = false;
      });
    },
    onBack: function onBack() {
      this.$emit('input', '');
      this.$emit('cancel');
    },
    onSearch: function onSearch(e) {
      e.preventDefault();
      this.$emit('search', this.value);
      return false;
    },
    onClickoutside: function onClickoutside() {
      this.isFocus = false;
      this.focusStatus = false;
    }
  }
};
