// _vant@0.11.5@vant/lib/mixins/popup/popup-manager.js
'use strict';

exports.__esModule = true;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _vue = require('vue');

var _vue2 = _interopRequireDefault(_vue);

var _Modal = require('./Modal');

var _Modal2 = _interopRequireDefault(_Modal);

var _popupContext = require('./popup-context');

var _popupContext2 = _interopRequireDefault(_popupContext);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var modalDefaultConfig = {
  className: '',
  customStyle: {}
};

var PopupManager = {
  getModal: function getModal() {
    var modal = _popupContext2.default.modal;


    if (!modal) {
      var ModalConstructor = _vue2.default.extend(_Modal2.default);
      modal = new ModalConstructor({
        el: document.createElement('div')
      });
      modal.$on('click', function () {
        PopupManager.handleOverlayClick();
      });

      _popupContext2.default.modal = modal;
    }

    return modal;
  },


  // close popup when click modal && closeOnClickOverlay is true
  handleOverlayClick: function handleOverlayClick() {
    var top = _popupContext2.default.top;

    if (top) {
      var instance = _popupContext2.default.instances[top.id];
      if (instance && instance.closeOnClickOverlay) {
        instance.close();
      }
    }
  },
  openModal: function openModal(config) {
    var id = config.id,
        dom = config.dom;

    var exist = _popupContext2.default.stack.some(function (item) {
      return item.id === id;
    });

    if (!exist) {
      var targetNode = dom && dom.parentNode && dom.parentNode.nodeType !== 11 ? dom.parentNode : document.body;
      _popupContext2.default.stack.push({ id: id, config: config, targetNode: targetNode });
      this.updateModal();
    };
  },
  closeModal: function closeModal(id) {
    var stack = _popupContext2.default.stack;


    if (stack.length) {
      if (_popupContext2.default.top.id === id) {
        stack.pop();
        this.updateModal();
      } else {
        _popupContext2.default.stack = stack.filter(function (item) {
          return item.id !== id;
        });
      }
    }
  },
  updateModal: function updateModal() {
    var modal = this.getModal();
    var el = modal.$el;

    if (el.parentNode) {
      el.parentNode.removeChild(el);
    }

    if (_popupContext2.default.top) {
      var _context$top = _popupContext2.default.top,
          targetNode = _context$top.targetNode,
          config = _context$top.config;


      targetNode.appendChild(el);
      (0, _assign2.default)(modal, (0, _extends3.default)({}, modalDefaultConfig, config));
    }
  }
};

exports.default = PopupManager;
