// _vant@0.11.5@vant/lib/uploader/index.js
"use strict";

exports.__esModule = true;
exports.default = {
  render: function render() {
    var _vm = this;var _h = _vm.$createElement;var _c = _vm._self._c || _h;return _c('div', { staticClass: "van-uploader" }, [_vm._t("default"), _c('input', _vm._b({ ref: "input", staticClass: "van-uploader__input", attrs: { "type": "file", "disabled": _vm.disabled }, on: { "change": _vm.onValueChange } }, 'input', _vm.$attrs, false))], 2);
  },

  name: 'van-uploader',

  props: {
    disabled: Boolean,
    beforeRead: Function,
    afterRead: Function,
    resultType: {
      type: String,
      default: 'dataUrl'
    }
  },

  methods: {
    onValueChange: function onValueChange(event) {
      var _this = this;

      if (this.disabled) {
        return;
      }

      var file = event.target.files[0];
      if (!file || this.beforeRead && !this.beforeRead(file)) {
        return;
      }

      var reader = new FileReader();
      reader.onload = function (e) {
        _this.afterRead && _this.afterRead({
          file: file,
          content: e.target.result
        });
        _this.$refs.input && (_this.$refs.input.value = '');
      };

      if (this.resultType === 'dataUrl') {
        reader.readAsDataURL(file);
      } else if (this.resultType === 'text') {
        reader.readAsText(file);
      }
    }
  }
};
